### PR TITLE
Refactor and enhance.

### PR DESCRIPTION
**Pull request by modestofiguereo**
Changes to be committed:
	modified:   .eslintrc
	modified:   index.js
	modified:   package.json

You can use the tool like this:
```
const createScenario = createTestSuite(api);

const scenarioDescription = {
    title: 'POST /bids-count; 400 response; if the required params (startDate, finalDate) are not passed.',
    path: '/bids-count',
    method: 'post',
    headers: { Authorization: `Bearer ${authToken}` },
    assertions: [
      (res, t) => t.equal(res.status, 400),
      (res, t) => t.equal(res.headers['content-type'], 'text/plain; charset=utf-8'),
    ],
 };
const dependencies =  {
    secrect: JWT_SECRET_KEY,
    countBidsDoneByNinjas: sinon.stub().returns(Promise.resolve(42)),
    countAllBids: sinon.stub().returns(Promise.resolve(100)),
 };

createScenario(scenarioDescription, dependencies);
```

### CODE REVIEW

**Comment by chrsalx** [comment]
In general I don't like the idea of the framework being the one who handles how you should work with your koa apps. Feels too opinionated. I prefer to only have a working listening server passed on the function and then work from there. Also this allows to work with basically any node http-server.

But I really like the idea of having multiple individual assertions instead of just one big expect object.

Also: I see no reason why we should use supertest-koa-agent which is a sub tool when supertest works just as well. Althought I cannot say the same about blue-tape, I like the dea of being able to allow generators as assertions.

File: index.js [hunk 1]
```diff
@@ -1,54 +1,38 @@
...
+    const assertions = config.assertions || [];
+
+    const app = apiFactory(dependencies);
+    let request = agent(app)[httpMethod](path)
+      .query(query || {})
+      .send(body || {});
+
+    request = Object.keys(headers).reduce(
+      (req, header) => req.set(header, headers[header]), request
+    );
```
Comment (chrsalx) [inline index.js]: why not just pass the headers object to the request?

`request.set` supports objects as arguments.

File: index.js [hunk 2]
```diff
@@ -1,54 +1,38 @@
...
-    put: request.put,
-  };
-  return methods[method.toLowerCase()];
-}
-
-function isGenerator(fn) {
-  return fn.constructor.name === 'GeneratorFunction';
-}
+const tape = require('blue-tape');
+const agent = require('supertest-koa-agent');
```
Comment (chrsalx) [inline index.js]: I see no reason why we should use supertest-koa-agent which is a sub tool when supertest works just as well. Althought I cannot say the same about blue-tape, I like the idea of being able to allow generators as assertions.

File: index.js [hunk 3]
```diff
@@ -1,54 +1,38 @@
...
-  function addScenario(config) {
-    scenarios.push(config);
-  }
-
-  function run() {
-    scenarios.forEach((scenario) => {
-      const req = convertScenarioToRequestPromise(request, scenario);
-      runTestForScenario(scenario, req);
+function addScenario(apiFactory, config, dependencies) {
+  function* bidsCount(t) {
```
Comment (chrsalx) [inline index.js]: missed a function name there, buddy. 

rename it to something related to this project. :p

File: index.js [hunk 4]
```diff
@@ -1,54 +1,38 @@
...
+
+    request.end((err) => {
+      t.error(err, 'test produced no errors.');
     });
-    return new Promise(resolve => test.onFinish(resolve));
   }
 
-  return { scenario: addScenario, run };
-};
+  tape(config.title, co.wrap(bidsCount));
```
Comment (chrsalx) [inline index.js]: tape-async would take away the need to include 'co' on this project.

**Comment by modestofiguereo** [comment]
I updated the PR with changes taking into account the comments you made.

**Pull request merged**